### PR TITLE
Possible fix for edge-case where realm is nil or false

### PR DESCRIPTION
**Pull request by linaori**

### CODE REVIEW

File: partyTracker.lua [hunk 1]
```diff
@@ -18,7 +24,6 @@ local function OnCharacterLeft(name, data)
 end
 
 local function updateParty()
-    local partyUpdated = false
```
Comment (linaori) [inline partyTracker.lua:21]: was unused

File: partyTracker.lua [hunk 2]
```diff
@@ -30,10 +35,11 @@ local function updateParty()
...
     -- Ensure existence of partyStack entity
     for i=1, GetNumGroupMembers() do
-        local fullName, realm, unitExists = util:GetNameRealm(prefix .. i)
-        -- print("fullName: " .. fullName)
-        if fullName ~= "Unknown" and fullName ~= prefix .. i and unitExists then -- If the user identifier cannot be fetched then we just omit (usually it happens when its us in the party)
-            local entryKey = util:GetCurrentRegion() .. "/" .. util:GetRealmSlug(realm) .. "/" .. fullName
+        local unitName = prefix .. i
+        local characterName, characterRealm = util:GetNameRealm(unitName)
+
+        if characterName and characterName ~= UNKNOWNOBJECT and characterName ~= unitName and characterRealm then -- If the user identifier cannot be fetched then we just omit (usually it happens when its us in the party)
```
Comment (linaori) [inline partyTracker.lua:41]: `unitExists` would've been the same as `prefix .. i`, thus always returned a truth-y value, causing the `if` to trigger even when it shouldn't. This argument has been removed from `GetNameRealm` as it was only used here. Now checks if the realm actually exists as without we can't do anything

File: partyTracker.lua [hunk 3]
```diff
@@ -48,14 +54,14 @@ local function updateParty()
...
-                local genderId = util:GenderLookup(fullName)
+                local _, _, raceId = UnitRace(characterName)
+                local _, _, classId = UnitClass(characterName)
+                local genderId = util:GenderLookup(characterName)
 
                 if raceId and classId and genderId then
                     partyStack[entryKey] = {
                         joined=lastTick, lastSeenAt=lastTick, 
-                        emittedLeftEvent=false, emitedJoinEvent=false,
+                        emittedLeftEvent=false, emittedJoinEvent=false,
```
Comment (linaori) [inline partyTracker.lua:64]: This was a typo that luckily didn't break anything

File: utils.lua [hunk 4]
```diff
@@ -50,74 +83,89 @@ function util:GetRealmSlug(realm)
     return realmSlug or realm
 end
 
-local unitIdentifiers = {"mouseover", "player", "target", "focus", "pet", "vehicle"}
+local unitIdentifiers = {
+    ["mouseover"] = true,
+    ["player"] = true,
```
Comment (linaori) [inline utils.lua:88]: This was used as a map, and is now working as one: `unitIdentifiers[identifier]`

File: utils.lua [hunk 5]
```diff
@@ -50,74 +83,89 @@ function util:GetRealmSlug(realm)
...
+        unitIdentifiers[k .. "target"] = true
     end
 end
 
--- Can be called with isUnit("Marahin"), IsUnit("party1"))
-function util:IsUnit(identifier, isPlayer)
-    if not isPlayer and type(identifier) == "string" and identifier:find("-", nil, true) then
-        isPlayer = true
+-- Can be called with isUnit("raid14"), IsUnit("party1"))
+function util:IsUnit(identifier)
```
Comment (linaori) [inline utils.lua:122]: This function is rewritten as the second argument was basically never used as `isPlayer` and caused a potential bug in other scenarios. Previously it also tried to resolve player names in `UnitExists` and `UnitIsPlayer`, but those only work with unit names, not player names.

File: utils.lua [hunk 6]
```diff
@@ -50,74 +83,89 @@ function util:GetRealmSlug(realm)
...
-        unit = arg1
         if unitIsPlayer then
-            name, realm = UnitName(unit)
-            realm = realm and realm ~= "" and realm or GetNormalizedRealmName()
+            characterName, characterRealm = UnitName(characterOrUnitName)
+            characterRealm = characterRealm ~= "" and characterRealm or GetNormalizedRealmName()
         end
 
-        return name, realm, unit
+        return characterName, characterRealm
```
Comment (linaori) [inline utils.lua:149]: The unit argument was not used except for the part where it was treated as `unitExists` while it returns the first argument of the function instead.

**Review by Marahin — APPROVED** [review]

**Comment by Marahin** [comment]
LGTM, let's ship the test release to internal testers, and if no bugs are reported, merge it in

**Pull request merged**